client/src/components/layout/MovieReviewForm.js

```diff
@@ -0,0 +1,131 @@
+import React, { useState } from "react"
+import { Redirect } from "react-router-dom"
+import translateServerErrors from "../../services/translateServerErrors.js";  
+import ErrorList from "./ErrorList.js"
+
+const MovieReviewForm = ({ movie, movieId }) => {
+
+  const [newReview, setNewReview] = useState({
+    content:"",
+    rating:"",
+    spoilerWarning: false
+  })
+  const [errors, setErrors] = useState([])
+  const [redirect, setRedirect] = useState(false)
+
+  const postReview = async (newReviewData) => {
+    try {
+      const response = await fetch(`/api/v1/movies/${movieId}/reviews`, {
+        method: "POST",
+        headers: new Headers({
+          "Content-Type": "application/json"
+        }),
+        body: JSON.stringify(newReviewData)
+      })
+      if (!response.ok) {
+        if (response.status === 422) {
+          const errorBody = await response.json()
+          const newErrors = translateServerErrors(errorBody.errors)
+          return setErrors(newErrors)
+        } else {
+          const errorMessage = `${response.status} (${response.statusText})`
+          const error = new Error(errorMessage)
+          throw error
+        }
+      } else {
+        const responseBody = await response.json()
+        setErrors([])
+        setRedirect(true)
+      }
+    } catch (error) {
+      console.error(`Error in fetch: ${error.message}`)
+    }
+  }
+
+  const handleInputChange = event => {
+    
+    setNewReview({
+      ...newReview,
+      [event.currentTarget.name]: event.currentTarget.value
+    })
+    if (event.currentTarget.name === "spoilerWarning") {
+      if (event.currentTarget.checked) {
+        setNewReview({
+          ...newReview,
+          [event.currentTarget.name]: true
+        });
+      }
+      else {
+        setNewReview({
+          ...newReview,
+          [event.currentTarget.name]: false,
+        });
+      }
+    }
+  }
+
+  const handleSubmit = (event) => {
+    event.preventDefault()
+    postReview(newReview)
+    clearForm()
+  }
+
+  const clearForm = () => {
+    setNewReview({
+      content:"",
+      rating:"",
+      spoilerWarning: false
+    })
+  }
+
+  if(redirect) {
+    return <Redirect to={{pathname:`/movies/${movieId}`}}/>
```

Comment:
We should not be redirecting to the same page after a form submission. Look back at the week 4 material to see how we handled this in the past. One of the key features of React is the render lifecycle that helps to prevent whole page refreshes - like default form submissions or redirects. After the form has submitted successfully the new review object should be added to state with the rest of the reviews. This setting of state causes the reviews on the show page to re-render and display the new review.